src/loop.jl

```diff
@@ -106,24 +106,17 @@ function interval_loop(
                     getfield(case, Symbol(p))[interval_start:interval_end, 2:end]
                 )
             end
-            for g in keys(sets.profile_resources_num_rep)
+            for g in values(sets.profile_resources_num_rep)
                 for h in 1:interval
-                    for i in
-                        1:length(
-                        sets.profile_resources_idx[sets.profile_resources_num_rep[g]]
-                    )
+                    for i in 1:length(sets.profile_resources_idx[g])
                         JuMP.set_normalized_rhs(
                             m[:profile_upper_bound][g, i, h],
-                            simulation_profile[sets.profile_to_group[sets.profile_resources_num_rep[g]]][
-                                h, i
-                            ],
+                            simulation_profile[sets.profile_to_group[g]][h, i],
```

Comment:
Not sure for the above line `m[:profile_upper_bound][g, i, h]` whether the `g` index is arbitrary, but since that is being changed, could someone confirm that is the case?

Reply:
So long as `JuMP` constraints are okay with using strings as indices, I think this looks right. The `g` index is not arbitrary, it is helping to specify the specific constraint that is being updated. We are essentially creating a constraint for each profile-based generator type (this is what `g` is referring to), each generator of each profile-based generator type, and each hourly time step.

Reply:
I guess it is the case if @jenhagg was able to run a PCM for Europe, no?